src/radical/pilot/session.py

```diff
@@ -626,16 +629,26 @@ def add_resource_config(self, resource_config):
 
         if isinstance(resource_config, str):
 
-            # let exceptions fall through
             rcs = ru.Config('radical.pilot.resource', name=resource_config)
+            domain = os.path.splitext(os.path.basename(resource_config))[0]
+            if domain not in self._rcfgs:
+                self._rcfgs[domain] = {}
 
             for rc in rcs:
-                self._log.info('load rcfg for %s' % rc)
-                self._rcfgs[rc] = rcs[rc].as_dict()
+                self._log.info('load rcfg for "%s.%s"' % (domain, rc))
+                self._rcfgs[domain][rc] = rcs[rc].as_dict()
 
         else:
-            self._log.debug('load rcfg for %s', resource_config.label)
-            self._rcfgs[resource_config.label] = resource_config.as_dict()
+
+            if not resource_config.label:
+                resource_config.label = RESOURCE_CONFIG_LABEL_DEFAULT
```

Comment:
I think we should raise an error here and require a label to be set.  Otherwise the user will not know how to address the cfg (it can be documented, but still), but also new configs will overwrite previous ones.  What do you think?

Reply:
To have a default value could be useful in some cases when the user keeps updating resource config run after run (load config, then add to the session, and repeats these steps) without creating a label. New configs (without a label) will overwrite the previous and that's ok, kinda we provide a place where no label config can be kept.

Reply:
How would the user reference the changed config in a pilot description if she does not know the label?  I would think requiring a label is fine - the user can still overwrite the cfg again and again...

Reply:
> How would the user reference the changed config in a pilot description if she does not know the label?

```python
from radical.pilot.session import RESOURCE_CONFIG_LABEL_DEFAULT
```
but this constant can be moved where all constants are and be retrieved just as `rp.RESOURCE_CONFIG_LABEL_DEFAULT` (and we always have some default values)

The point is that the user who doesn't know about labels will not get its own config either way (either as you propose with an exception or later without knowing which resource label to set), but the advanced user can get some benefits out of it by eliminating extra operations
```python
...
for cfg in cfg_list:  # configs could have different LM, RM, etc.
    session.add_resource_config(resource_config=cfg)
    pd = {'resource': rp.RESOURCE_CONFIG_LABEL_DEFAULT}
...
```
I mean just not to limit this possibility

Reply:
I don't see this ever being used...  I mean, it does not hurt either, so `¯\_(ツ)_/¯`